flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/UnionInputGate.java

```diff
@@ -118,7 +118,7 @@ public UnionInputGate(InputGate... inputGates) {
 			}
 
 			if (!inputGatesWithData.isEmpty()) {
-				isAvailable = AVAILABLE;
+				availabilityHelper.getUnavailableToResetAvailable();
```

Comment:
Calling `getUnavailableToResetAvailable` without getting the result, is a bit confusing. Maybe we should provide an additional helper method like `resetAvailable` or `markAvailable` that does the same thing but returns `void`?

Reply:
Yes, that makes sense.